### ISSUE TITLE
Arcticdb fails to write to S3 if the "region" is specified in the URL options

### ISSUE THREAD

**Issue by alokwatve** [man-group/ArcticDB#261]
S3 writes work when the region is part of the URL e.g., The following URL works:
https://s3**.us-east-1.**amazonaws.com/.&sa=D&source=docs&<more options> 

However, when S3 region is specified in the options, the write fails:
https://s3.amazonaws.com?**region=us-east-1**&sa=D&source=docs&<more_options>

As far as we know both the ways are valid for S3 access.

**Comment by mehertz** [comment]
Marked as a bug. We'll try and replicate as a first step.

**Comment by jjerphan** [comment]
Hi @alokwatve, thank you for this report!

Could you [craft a minimal bug report](https://matthewrocklin.com/minimal-bug-reports)? This would allow ArcticDB's maintainers to resolve your problem more effectively. Thank you.

**Comment by poodlewars** [comment]
We should support standard S3 URIs. We seem to have ended up with our own custom URI specification on our `Arctic` API.

**Comment by hk1997** [comment]
Hi is this issue still up for grabs?
I'm new to the project would like to start with something.

**Comment by poodlewars** [comment]
> Hi is this issue still up for grabs? I'm new to the project would like to start with something.

Yep absolutely, should be a good one to get started with. Just a couple of things to bear in mind,

- You need to sign off your commits in a special way https://github.com/man-group/ArcticDB/blob/master/docs/mkdocs/docs/technical/contributing.md 
- Ideally it would be good to have some user friendly validation of the URIs (see #790 )
- We still need to keep backwards compatibility with our current (custom) S3 URI scheme

**Comment by hk1997** [comment]
Thanks Alex,
From what I understood going through the codebase, the file which might require change should be: https://github.com/man-group/ArcticDB/blob/master/python/arcticdb/adapters/s3_library_adapter.py

Also how do you validate your changes, is there a testing environment where we can test upload etc?

**Comment by poodlewars** [comment]
That's right @hk1997 , the changes should be very localized there.

Our unit tests in `test_arctic.py` test against S3 storages using a simulator. This is set up with parameterized fixtures like this https://github.com/man-group/ArcticDB/blob/f32e342aeaf598e36a3ef20c5e0364a9b8956975/python/tests/conftest.py#L180 
. You'll notice there are also some tests that run against real storages, which runs in our CI against our AWS account.

**Comment by hk1997** [comment]
@poodlewars Can you please share some instructions to setup the project on mac os and run unit tests.

**Comment by ar6100** [comment]
I couldn't find any information about this, but would resolving this issue also enable the use of S3 Express Zone One? It's probably user error on my part, but I couldn't get it to work, and I suspect it's related to the S3 connection URI issue.
